### ISSUE TITLE
Image src binding in Vue

### ISSUE THREAD

**Issue by charlie-harvard** [susanBuck/e28-fall19#95]
I couldn't display my images using the "Comprehensive" method in the `App.vue` file. I was migrating my basic Vue scripts to Vue CLI.

```html
<img :src=getImg('apple')>
```
The data variable in `App.vue` was set like this:
```javascript
...
data: function() {
    return {
      iconImg:  {
          id: "0",
          img: "./assets/apple.png"
      }
}
...
```
Then I found the reason and here's my modified code.
Reference: [Asset URL Handling](https://vue-loader.vuejs.org/guide/asset-url.html)
```javascript
...
data: function() {
    return {
      iconImg:  {
          id: "0",
          img: require("./assets/apple.png")
      }
}
...
```

**Comment by susanBuck** [comment]
Thanks for sharing @charlie-harvard. The require approach came up in this recent thread: #107.

Note, though, that it should still be possible to load images in SFC's without using require.

For reference, here's an example from ZipFoods (Week 9) where I loaded an image from `App.vue`: 
```
<img id='logo' alt='ZipFoods logo' src='./assets/images/zipfoods-logo.png' />
```

Not sure why it wasn't working on your end, but glad you found a solution! If you decide to try it again without require and can't get it to function, feel free to re-open this thread or start a new one!

**Comment by extra808** [comment]
When using `npm run serve`, the ZipFoods logo in the App.vue changes from 
```HTML
<img id='logo' alt='ZipFoods logo' src='./assets/images/zipfoods-logo.png' />
```
to this in the browser:
```HTML
<img id="logo" alt="ZipFoods logo" src="/img/zipfoods-logo.a23a3a29.png">
```
Something is altering that img src path to a virtual location, /img/. But the same is not occurring for the img src in ShowProduct.vue, the path for the first product image in the browser is "./../assets/images/products/1.jpg"

I thought maybe it would only be an issue using `npm run serve` but I tried `npm run build` and it also did not copy the product/ images to a sub-folder within dist/.

**Comment by susanBuck** [comment]
@extra808 I just cloned your e28 repository and ran your zipfoods app locally.

In your `ShowProduct.vue`, here's how I wrote the image path (without using require):

```html
<img
            class='product-thumb'
            :src='"./../assets/images/products/" + product.id + ".jpg"'
            :alt='product.name'
        />
```

This compiled w/o issue, and the image loads, pulling from the virtual location:

![image](https://user-images.githubusercontent.com/237711/68609763-333c3780-0484-11ea-861e-f77a3225e354.png)

Can you test the exact path as I showed above on your end and report back?

Trying to figure out why we're having different outcomes.

**Comment by extra808** [comment]
Using `:src='"./../assets/images/products/" + product.id + ".jpg"'` works. I was trying to avoid messing with Vetur's defaults so I kept the double quotes on the outside, which does not work: `:src="'./../assets/images/products/' + product.id + '.jpg'"` doesn't through an error, the asset path just isn't converted. I didn't notice this before but if I don't have nested quotes or concatenation at all (i.e. `"./../assets/images/products/1.jpg"`), npm run serve throws errors:

```
 error  in ./src/components/ShowProduct.vue?vue&type=template&id=1acef832&

Module Error (from ./node_modules/vue-loader/lib/loaders/templateLoader.js):
(Emitted value instead of an instance of Error) 

  Errors compiling template:

  invalid expression: Unexpected token '.' in

    ./../assets/images/products/1.jpg

  Raw expression: :src="./../assets/images/products/1.jpg"
```
The unexpected token also leads to an "'v-bind' directives require an attribute value" error. This still happens if I replace the double quotes with single quotes. If I _nest_ the quotes (doesn't matter whether double quotes are on the inside or outside), then it works.

Either `src="./assets/images/zipfoods-logo.png"` or `src='./assets/images/zipfoods-logo.png'` works in App.vue apparently because v-bind isn't involved; if I change it to `:src` I get the same unexpected token error with it.

In summary, these seem to be the rules for asset paths to be built with v-bind:

1. they must have nested quotes and 
1. if variables are used to build the path, the outer quotes must be single quotes

Therefore, Vetur's default for "fixing" quotes are a problem.

**Comment by susanBuck** [comment]
@extra808 - Ok, I'm following you now. Thanks for the extra details! 

Amended my other post (#107) based on these findings.
